ystia/yorc
# Feature/gh 183 migrate to go modules

# Pull Request description

## Description of the change

This PR migrate Yorc's dependencies management system from govendor to [official Go modules](https://github.com/golang/go/wiki/Modules)

### What I did

* Converted govendor file to `go.mod` using standard go toolchain
* removed vendor directory
* use go.mod to identify tools versions (will allow common, reproductible builds)
* removed use of go-bindata to ship resources in Yorc binary, use the same system than the bootstrap process. Not having those generated content will ease plugins development
* document plugins, how they work and how to create a plugin  

### How to verify it

Make sure you have Go 1.11+ installed.
Run:

``` bash
make tools
make
```

It should produce a binary. Test it.

Feel free to test the plugin how to guide. - Temporary built documentation: https://yorc.readthedocs.io/en/feature-gh-183_migrate_to_go_modules/plugins.html) - One notable change is as 3.2.0-M2 is not yet available you should replace `go get -m github.com/ystia/yorc/v3@v3.2.0-M2` by `go get -m github.com/ystia/yorc/v3@v3.0.0-20190211161143-466e3088f98f` 

### Description for the changelog

* Move to Go modules to manage dependencies ([GH-183](https://github.com/ystia/yorc/issues/183))
* Document How to create a Yorc Plugin ([GH-119](https://github.com/ystia/yorc/issues/119))

## Applicable Issues

Fixes #183 

Also fixes #119
